File: python/thread/serial_sample.py
#!/usr/bin/python
# -*- coding: utf-8 -*-

# url
## http://ja.pymotw.com/2/threading/
## http://docs.python.jp/2.7/library/threading.html

##############
## zip fileを解凍する
##############

import threading
import glob
import time
import zipfile

path = "/home/kzk/downloads/*.zip"
zipfiles = glob.glob(path)

## reading onley unzip
def unzip(zfin):
    zf = zipfile.ZipFile(zfin, "r")
    for f in zf.namelist():  # zipは複数ファイルが１つにまとめられている前提のため
        #print "unzip", f
        zf.read(f)  # reading only
    zf.close

## non-thread for comparison
st = time.time()
for zf in zipfiles:
    unzip(zf)
    
et = time.time()
print "total execution time without threading: ", (et - st), "[s]"

# total execution time without threading:  57.6119029522 [s]

File: ruby/thread/serial_sample.rb
#!/usr/bin/ruby
# -*- coding: utf-8 -*-

require "rubygems"
require "zipruby"

# url
## http://www.namaraii.com/rubytips/?%E3%82%B9%E3%83%AC%E3%83%83%E3%83%89
## http://techracho.bpsinc.jp/piichan1031/2010_07_05/2030
## 

##############
## zip fileを解凍する
##############

## unzip function
def unzip(path)
  Zip::Archive.open(path) do |archives|
    archives.each do |a|
      unless a.directory?
        ## reading only
        a.read
      end
    end
  end
end

## non-thread for comparison
path = "/home/kzk/downloads/*.zip"
st = Time.now
for fpath in Dir.glob(path)
  unzip(fpath)
end
et = Time.now
puts "total execution time without threading: #{et - st} [s]"

# total execution time with threading: 54.494340185 [s]
